src/main/java/net/coderbot/iris/shaderpack/ShaderPackConfig.java

```diff
@@ -0,0 +1,79 @@
+package net.coderbot.iris.shaderpack;
+
+import static net.coderbot.iris.Iris.SHADERPACK_DIR;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+public class ShaderPackConfig {
+
+	public final Path shaderPackConfigPath;
+	private final Properties configProperties;
+	private final String comment;
+
+	private final Map<String, Option<Boolean>> booleanOptions = new HashMap<>();
+	private final Map<String, Option<Float>> floatOptions = new HashMap<>();
+	private final Map<String, Option<Integer>> integerOptions = new HashMap<>();
+
+	public ShaderPackConfig(String name) {
+		//optifine uses txt files, so we should do the same
+		shaderPackConfigPath = SHADERPACK_DIR.resolve(name + ".txt");
+		configProperties = new Properties();
+		comment = "This file stores the shaderpack configuration for the shaderpack " + name;
+	}
+
+	public Properties getConfigProperties() {
+		return configProperties;
+	}
+
+	//this 3 methods below should be used by the gui to get the available options and then use them
+	public Map<String, Option<Boolean>> getBooleanOptions() {
+		return booleanOptions;
+	}
+
+	public Map<String, Option<Float>> getFloatOptions() {
+		return floatOptions;
+	}
+
+	public Map<String, Option<Integer>> getIntegerOptions() {
+		return integerOptions;
+	}
+
```

Comment:
Considering that the fields are anyway final and that the getters don't do anything other than return each map, are getters really necessary?

Reply:
It's more of a personal preference, but It's not a big deal

Reply:
Also, I've been thinking, there wasn't any need to even organize it as a map, since the key to value mapping is stored as the option's name to the option itself, which seems redundant as you won't need it anyway. I think it should probably be converted to a HashSet.

Reply:
However, the values in the set/map shouldn't be modified either, so maybe add a getter that returns an immutable copy of the map, to make sure it's not tampered with

Reply:
I think if you want to do it that way, Immutable copies seems bit inefficient... What I'd recommend is have a method `Option<Integer> getIntegerOption(String name)` possibly with a method `Iterable<String> getIntegerOptions()` or something along those lines.

Reply:
Okay I just finished adding this, so can you do a quick update on your branch?

Reply:
I think you may have misunderstood, there seems to no longer be a method for getting an option by key. I suggested adding two methods for each option type, one for getting by key and another for getting all available keys

Reply:
Oh sorry about that, will fix now.